### PR TITLE
Update EffCancelEvent

### PR DESCRIPTION
**Pull request by Efnilite**
### Description
Adds error for trying to cancel the swim toggle event and updates `EffCancelEvent`:

- Added test
- Updated formatting

---

**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** #7054

### CODE REVIEW

File: src/main/java/ch/njol/skript/effects/EffCancelEvent.java [hunk 1]
```diff
@@ -42,65 +13,82 @@
...
+		"\tvictim is a player",
+		"\tvictim has the permission \"skript.god\"",
+		"\tcancel the event"
+})
 @Since("1.0")
 public class EffCancelEvent extends Effect {
+
 	static {
-		Skript.registerEffect(EffCancelEvent.class, "cancel [the] event", "uncancel [the] event");
+		Skript.registerEffect(EffCancelEvent.class, "[:un]cancel [the] event");
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/effects/EffCancelEvent.java:39]: why not two patterns :(

**Comment by cheeezburga** [comment]
Sorry! I misclicked

File: src/test/skript/junit/EffCancelEvent.sk [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	if event is not cancelled:
+		complete objective "event not cancelled by default" for {_test}
+
+	cancel event
+	if event is cancelled:
+		complete objective "cancel event" for {_test}
+
+	uncancel event
+	if event is not cancelled:
+		complete objective "uncancel event" for {_test}
```
Comment (Pikachu920) [inline src/test/skript/junit/EffCancelEvent.sk:21]: new line at end of file pls

File: src/test/skript/junit/EffCancelEvent.sk [hunk 3]
```diff
@@ -0,0 +1,21 @@
+test "SimpleJUnitTest" when running JUnit:
```
Comment (Pikachu920) [inline src/test/skript/junit/EffCancelEvent.sk:1]: could you please also add a test for the swim event error (maybe the player login one too if you want)? you can use the parse structure to ensure it errors as expected

**Review by Pikachu920 — CHANGES_REQUESTED** [review]

**Comment by Efnilite** [comment]
Is awaiting merge of #7066

File: src/main/java/ch/njol/skript/effects/EffCancelEvent.java [hunk 4]
```diff
@@ -62,19 +43,26 @@ public class EffCancelEvent extends Effect {
...
-		if (es == null)
+		Class<? extends Event>[] currentEvents = getParser().getCurrentEvents();
+
+		if (currentEvents == null)
 			return false;
-		for (final Class<? extends Event> e : es) {
-			if (Cancellable.class.isAssignableFrom(e) || BlockCanBuildEvent.class.isAssignableFrom(e))
+
+		if (cancel && getParser().isCurrentEvent(EntityToggleSwimEvent.class))
+			Skript.error("Cancelling a toggle swim event has no effect", ErrorQuality.SEMANTIC_ERROR);
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/effects/EffCancelEvent.java:59]:
```suggestion
			Skript.error("Cancelling a toggle swim event has no effect");
```

**Review by sovdeeth — CHANGES_REQUESTED** [review]

**Review by sovdeeth — APPROVED** [review]

**Review by Pikachu920 — APPROVED** [review]
woooooooooooooooooooo

File: src/main/java/ch/njol/skript/effects/EffCancelEvent.java [hunk 5]
```diff
@@ -42,39 +13,56 @@
...
-	
-	@SuppressWarnings("null")
+
 	@Override
-	public boolean init(final Expression<?>[] vars, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
+	public boolean init(Expression<?>[] expressions, int matchedPattern,
+						Kleenean isDelayed, ParseResult parseResult) {
 		if (isDelayed == Kleenean.TRUE) {
-			Skript.error("Can't cancel an event anymore after it has already passed", ErrorQuality.SEMANTIC_ERROR);
+			Skript.error("Can't cancel an event after it has already passed", ErrorQuality.SEMANTIC_ERROR);
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/effects/EffCancelEvent.java:48]:
```suggestion
			Skript.error("An event cannot be cancelled after it has already passed", ErrorQuality.SEMANTIC_ERROR);
```
I think this is a better statement to make

File: src/main/java/ch/njol/skript/effects/EffCancelEvent.java [hunk 6]
```diff
@@ -42,39 +13,56 @@
...
-		if (es == null)
+		Class<? extends Event>[] currentEvents = getParser().getCurrentEvents();
+
+		if (currentEvents == null)
 			return false;
-		for (final Class<? extends Event> e : es) {
-			if (Cancellable.class.isAssignableFrom(e) || BlockCanBuildEvent.class.isAssignableFrom(e))
+
+		if (cancel && getParser().isCurrentEvent(EntityToggleSwimEvent.class))
+			Skript.error("Cancelling a toggle swim event has no effect");
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/effects/EffCancelEvent.java:59]: Should this be a warning? Or do we want to return false here?

**Review by APickledWalrus — APPROVED** [review]

**Pull request merged**